### PR TITLE
wayland: keyboard: Cache text input parameters.

### PR DESCRIPTION
**Pull request by sulix**
Some applications (and embarrassingly, testime is one of them) call SDL_StartTextInput() or SDL_SetTextInputRect() every frame. On KDE/KWin with fcitx5, this causes there to be several preedit events every frame (particularly given some of the workarounds in Wayland_StartTextInput), which slows testime down to an unusable crawl.

Instead, make SDL_StartTextInput() a no-op if text input is already enabled, and cache the input rect, only changing it when the new rect is actually different.

With these changes, we only get preedit events (and hence SDL_TEXTEDITING events) when the preedit string actually changes. This matches the behaviour under XWayland, and works very smoothly.

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

**Review by flibitijibibo — APPROVED** [review]
Gets my approval, I think we're in a freeze so I'll wait for @slouken's approval as well.

**Pull request merged**
